### ISSUE TITLE
Visual Block Modes Doesn't Comment Correct Lines

### ISSUE THREAD

**Issue by RRethy** [terrortylor/nvim-comment#22]
```lua
local foo = 'foo'
local bar = 'bar'
local baz = 'baz'
local foo = 'foo'
local bar = 'bar'
local baz = 'baz'
```

1. Go into visual line mode on the first three lines and comment them out with `gc`
2. Undo with `u`
3. Go into visual block mode over all six lines and comment them out with `gc`

Only the first three lines will be commented out after the above steps when all 6 lines should be commented out.

**Comment by terrortylor** [comment]
I'm unable to replicate this, I've assumed that the file type is lua here and tried to replicate your behaviour.

Can you provide any more details on replicating this? Like a minimal vimrc (be it VimL or Lua)? Or even better replicate the behaviour in one of the integration tests?

**Comment by RRethy** [comment]
Minimal `init.lua` with nothing else loaded:

```lua
vim.opt.packpath = 'whatever is your packpath'
vim.cmd('packadd nvim-comment')
require('nvim_comment').setup()
```

In that same file do:

```
gg0Vjgcugg0<c-v>jjgc
```

All three lines should be commented, but only the first two are.

**Comment by terrortylor** [comment]
Awesome! I can replicate!
let me take a look

**Comment by terrortylor** [comment]
Closing as fix pushed and not heard back, re-open if still an issue though

**Comment by RRethy** [comment]
I don't use this plugin (at the time I was shopping around for a vim-commentary replacement but ended up sticking with tpope's plugin) so I don't know if the issue persists. In the future though I would recommend add a `Fix #22` in the commit message that fixes this issue so whoever made the issues gets notified of a fix (and the issue closes automatically) and it can be traced.
